cogs/events.py

```diff
@@ -1,21 +1,54 @@
 import traceback
 
-from discord import Color, Embed, Message
+from discord import Color, Embed, Message, Guild, Member
 from discord.ext.commands import Bot, Cog
 
+import re
+import aiohttp
+
 
 class Custom(Cog):
     def __init__(self, bot: Bot) -> None:
         self.bot = bot
 
+    @staticmethod
+    def get_link_code(string: str) -> str:
+        return string.split("/")[-1]
+
+    @staticmethod
+    async def check_our_invite(full_link: str, guild: Guild) -> bool:
+        guild_invites = await guild.invites()
+        for invite in guild_invites:
+            # discord.gg/code resolves to https://discordapp.com/invite/code after using session.get(invite)
+            if Custom.get_link_code(invite.url) == Custom.get_link_code(full_link):
+                return True
+        return False
+
     @Cog.listener()
     async def on_message(self, message: Message) -> None:
         # Check that the message is not from bot account
         if message.author.bot:
             return
 
-        if message.content.lower().startswith("help"):
-            await message.channel.send("Hey! Why don't you run the help command with `>>help`")
+        elif not isinstance(message.author, Member):
+            return
+
+        elif message.author.guild_permissions.administrator:
+            return
+
+        if "https:" in message.content or "http:" in message.content:
+            base_url = re.findall(r"http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*(),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+", message.content)
```

Comment:
We should be able to use a bit more refined regex pattern, this one finds all URLs but we only really need to check for discord URLs.

*Note: If you don't know that much about regex, you can contact me on discord*

Reply:
Don't mark this as resolved if you haven't actually resolved it...

Reply:
But, if someone uses url shortener to send a link, The what r u gonna Do?

Reply:
Not accept it, URL shorteners are unacceptable for this and if he did, your condition on line 58 wouldn't pass anyway, so it doesn't matter.